packages/components/src/tabs/tab.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import * as Ariakit from '@ariakit/react';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { TabProps } from './types';
+import warning from '@wordpress/warning';
+import { TabsContext } from './context';
+import { TabButton } from './styles';
+
+function Tab( {
+	children,
+	id,
+	className,
+	disabled,
+	icon,
```

Comment:
It does feel like it would be more ergonomic if we just always used `children`, but I also agree with your second thought about being aligned with `Button` and being able to easily leverage the tooltip that provides. The current implementation feels like the lesser of two evils to me.

Reply:
Now that I'm thinking about it, I'm not sure that `Tab` should be the component responsible for providing icon / tooltip support. That is not a specific feature of the Tabs pattern, but more of a generic button functionality (and in fact, we support it through the `Button` component).

Given that we're pushing for this more granular approach, it seems only reasonable that we allow consumers of `Tabs` to bring their own tab button implementation. Basically, we could expose the `render` prop directly — so that consumers of `Tab` could use `Button` in case they wanted it. We'd just need to ensure that:

- Styles still work correctly
- there are no incompatibilities in case users of `Button` specified props like `__experimentalIsFocusable` (for keyboard focus management)

<details>

<summary>Something like this</summary>

```diff
diff --git a/packages/components/src/tabs/stories/index.story.tsx b/packages/components/src/tabs/stories/index.story.tsx
index 9f79c57b5c..684a55b662 100644
--- a/packages/components/src/tabs/stories/index.story.tsx
+++ b/packages/components/src/tabs/stories/index.story.tsx
@@ -32,15 +32,9 @@ const Template: StoryFn< typeof Tabs > = ( props ) => {
 	return (
 		<Tabs { ...props }>
 			<Tabs.TabList>
-				<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
-					Tab 1
-				</Tabs.Tab>
-				<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
-					Tab 2
-				</Tabs.Tab>
-				<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-					Tab 3
-				</Tabs.Tab>
+				<Tabs.Tab id={ 'tab1' }>Tab 1</Tabs.Tab>
+				<Tabs.Tab id={ 'tab2' }>Tab 2</Tabs.Tab>
+				<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 			</Tabs.TabList>
 			<Tabs.TabPanel id={ 'tab1' }>
 				<p>Selected tab: Tab 1</p>
@@ -61,15 +55,11 @@ const DisabledTabTemplate: StoryFn< typeof Tabs > = ( props ) => {
 	return (
 		<Tabs { ...props }>
 			<Tabs.TabList>
-				<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' } disabled={ true }>
+				<Tabs.Tab id={ 'tab1' } disabled={ true }>
 					Tab 1
 				</Tabs.Tab>
-				<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
-					Tab 2
-				</Tabs.Tab>
-				<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-					Tab 3
-				</Tabs.Tab>
+				<Tabs.Tab id={ 'tab2' }>Tab 2</Tabs.Tab>
+				<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 			</Tabs.TabList>
 			<Tabs.TabPanel id={ 'tab1' }>
 				<p>Selected tab: Tab 1</p>
@@ -89,9 +79,24 @@ const WithTabIconsAndTooltipsTemplate: StoryFn< typeof Tabs > = ( props ) => {
 	return (
 		<Tabs { ...props }>
 			<Tabs.TabList>
-				<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' } icon={ wordpress } />
-				<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' } icon={ link } />
-				<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' } icon={ more } />
+				<Tabs.Tab
+					id={ 'tab1' }
+					render={
+						<Button icon={ wordpress } label="Tab 1" showTooltip />
+					}
+				/>
+				<Tabs.Tab
+					id={ 'tab2' }
+					render={
+						<Button icon={ link } label="Tab 2" showTooltip />
+					}
+				/>
+				<Tabs.Tab
+					id={ 'tab3' }
+					render={
+						<Button icon={ more } label="Tab 3" showTooltip />
+					}
+				/>
 			</Tabs.TabList>
 			<Tabs.TabPanel id={ 'tab1' }>
 				<p>Selected tab: Tab 1</p>
@@ -119,15 +124,9 @@ const UsingSlotFillTemplate: StoryFn< typeof Tabs > = ( props ) => {
 		<SlotFillProvider>
 			<Tabs { ...props }>
 				<Tabs.TabList>
-					<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
-						Tab 1
-					</Tabs.Tab>
-					<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
-						Tab 2
-					</Tabs.Tab>
-					<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-						Tab 3
-					</Tabs.Tab>
+					<Tabs.Tab id={ 'tab1' }>Tab 1</Tabs.Tab>
+					<Tabs.Tab id={ 'tab2' }>Tab 2</Tabs.Tab>
+					<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 				</Tabs.TabList>
 				<Fill name="tabs-are-fun">
 					<Tabs.TabPanel id={ 'tab1' }>
@@ -181,15 +180,9 @@ const CloseButtonTemplate: StoryFn< typeof Tabs > = ( props ) => {
 							} }
 						>
 							<Tabs.TabList>
-								<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
-									Tab 1
-								</Tabs.Tab>
-								<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
-									Tab 2
-								</Tabs.Tab>
-								<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-									Tab 3
-								</Tabs.Tab>
+								<Tabs.Tab id={ 'tab1' }>Tab 1</Tabs.Tab>
+								<Tabs.Tab id={ 'tab2' }>Tab 2</Tabs.Tab>
+								<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 							</Tabs.TabList>
 							<Button
 								variant={ 'tertiary' }
@@ -242,17 +235,11 @@ const ControlledModeTemplate: StoryFn< typeof Tabs > = ( props ) => {
 				} }
 			>
 				<Tabs.TabList>
-					<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
-						Tab 1
-					</Tabs.Tab>
+					<Tabs.Tab id={ 'tab1' }>Tab 1</Tabs.Tab>
 
-					<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
-						Tab 2
-					</Tabs.Tab>
+					<Tabs.Tab id={ 'tab2' }>Tab 2</Tabs.Tab>
 
-					<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-						Tab 3
-					</Tabs.Tab>
+					<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 				</Tabs.TabList>
 				<Tabs.TabPanel id={ 'tab1' }>
 					<p>Selected tab: Tab 1</p>
@@ -311,19 +298,11 @@ const TabBecomesDisabledTemplate: StoryFn< typeof Tabs > = ( props ) => {
 			</Button>
 			<Tabs { ...props }>
 				<Tabs.TabList>
-					<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
-						Tab 1
-					</Tabs.Tab>
-					<Tabs.Tab
-						id={ 'tab2' }
-						title={ 'Tab 2' }
-						disabled={ disableTab2 }
-					>
+					<Tabs.Tab id={ 'tab1' }>Tab 1</Tabs.Tab>
+					<Tabs.Tab id={ 'tab2' } disabled={ disableTab2 }>
 						Tab 2
 					</Tabs.Tab>
-					<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-						Tab 3
-					</Tabs.Tab>
+					<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 				</Tabs.TabList>
 				<Tabs.TabPanel id={ 'tab1' }>
 					<p>Selected tab: Tab 1</p>
@@ -353,17 +332,9 @@ const TabGetsRemovedTemplate: StoryFn< typeof Tabs > = ( props ) => {
 			</Button>
 			<Tabs { ...props }>
 				<Tabs.TabList>
-					{ ! removeTab1 && (
-						<Tabs.Tab id={ 'tab1' } title={ 'Tab 1' }>
-							Tab 1
-						</Tabs.Tab>
-					) }
-					<Tabs.Tab id={ 'tab2' } title={ 'Tab 2' }>
-						Tab 2
-					</Tabs.Tab>
-					<Tabs.Tab id={ 'tab3' } title={ 'Tab 3' }>
-						Tab 3
-					</Tabs.Tab>
+					{ ! removeTab1 && <Tabs.Tab id={ 'tab1' }>Tab 1</Tabs.Tab> }
+					<Tabs.Tab id={ 'tab2' }>Tab 2</Tabs.Tab>
+					<Tabs.Tab id={ 'tab3' }>Tab 3</Tabs.Tab>
 				</Tabs.TabList>
 				<Tabs.TabPanel id={ 'tab1' }>
 					<p>Selected tab: Tab 1</p>
diff --git a/packages/components/src/tabs/styles.ts b/packages/components/src/tabs/styles.ts
index 468900356e..6f35b276fb 100644
--- a/packages/components/src/tabs/styles.ts
+++ b/packages/components/src/tabs/styles.ts
@@ -2,11 +2,11 @@
  * External dependencies
  */
 import styled from '@emotion/styled';
+import * as Ariakit from '@ariakit/react';
 
 /**
  * Internal dependencies
  */
-import Button from '../button';
 import { COLORS } from '../utils';
 import { space } from '../ui/utils/space';
 import { reduceMotion } from '../utils/reduce-motion';
@@ -20,7 +20,7 @@ export const TabListWrapper = styled.div`
 	}
 `;
 
-export const TabButton = styled( Button )`
+export const Tab = styled( Ariakit.Tab )`
 	&& {
 		position: relative;
 		border-radius: 0;
diff --git a/packages/components/src/tabs/tab.tsx b/packages/components/src/tabs/tab.tsx
index 0406d80bde..05c32d83ad 100644
--- a/packages/components/src/tabs/tab.tsx
+++ b/packages/components/src/tabs/tab.tsx
@@ -15,17 +15,9 @@ import { useContext } from '@wordpress/element';
 import type { TabProps } from './types';
 import warning from '@wordpress/warning';
 import { TabsContext } from './context';
-import { TabButton } from './styles';
+import { Tab as StyledTab } from './styles';
 
-function Tab( {
-	children,
-	id,
-	className,
-	disabled,
-	icon,
-	title,
-	style,
-}: TabProps ) {
+function Tab( { children, id, className, disabled, render, style }: TabProps ) {
 	const context = useContext( TabsContext );
 	if ( ! context ) {
 		warning( '`Tabs.TabList` must be wrapped in a `Tabs` component.' );
@@ -34,22 +26,16 @@ function Tab( {
 	const { store, instanceId } = context;
 	const instancedTabId = `${ instanceId }-${ id }`;
 	return (
-		<Ariakit.Tab
+		<StyledTab
 			store={ store }
 			id={ instancedTabId }
 			className={ className }
 			style={ style }
 			disabled={ disabled }
-			render={
-				<TabButton
-					icon={ icon }
-					label={ icon ? title : undefined }
-					showTooltip={ true }
-				/>
-			}
+			render={ render }
 		>
 			{ children }
-		</Ariakit.Tab>
+		</StyledTab>
 	);
 }
 
diff --git a/packages/components/src/tabs/types.ts b/packages/components/src/tabs/types.ts
index 18e5a19d3a..83c79e632e 100644
--- a/packages/components/src/tabs/types.ts
+++ b/packages/components/src/tabs/types.ts
@@ -118,6 +118,7 @@ export type TabProps = {
 	 * @default false
 	 */
 	disabled?: boolean;
+	render?: React.ReactElement;
 };
 
 export type TabPanelProps = {

```

</details>